### PR TITLE
[doc] Manifest Annotations

### PR DESCRIPTION
**Pull request by pkriens**
- Added shortcuts for MavenBndRepository GAVs
- Explained filter generation with the @Requirement

Signed-off-by: Peter Kriens <Peter.Kriens@aqute.biz>

### CODE REVIEW

File: docs/_chapters/230-manifest-annotations.md [hunk 1]
```diff
@@ -72,6 +71,26 @@ Require-Capability: \
 
 The invisible link created between user code and the indirect requirement is a powerful mechanism that enables automatic validation of a bundle closure.
 
+The actual requirement `filter:` directive is constructed from an `AND` of the `filter()`, `name()`, and `version()` annotation methods. All fields are optional. The name field will create an assertion that the given namespace equals the value of the `name()` annotation method. For example, if the namespace is `com.example.foo` and the `name()` method has the value `bar` then the filter is `(com.example.foo=bar)`. If a version is specified, it will be expanded to a filtered version-range expression.  The convention of using the namespace name as the property key is commonly used in OSGi specification. For example, the filter `(osgi.wiring.package=com.example.foo)` is the filter for an Import-Package `com.example.foo` while `osgi.wiring.package` is the namespace for the packages.
+
+The filter will **not** be processed for macros like the `@Header` annotation is. The filter can therefore not refer to any annotation methods in meta-annotated classes.
```
Comment (bjhargrave) [inline docs/_chapters/230-manifest-annotations.md]: This is not entirely true. See https://bnd.bndtools.org/chapters/235-accessor-properties.html.

**Comment by pkriens** [comment]
I spent half an hour going over the `getFilter()` method and I fail to see where -any- macro processing takes place on the filter. Am I missing something?

File: docs/_chapters/230-manifest-annotations.md (hunk 1 above)
Comment (pkriens) [reply docs/_chapters/230-manifest-annotations.md]: I spent half an hour going over the getFilter() method and I fail to see where -any- macro processing takes place on the filter. Am I missing something?

Comment (rotty3000) [reply docs/_chapters/230-manifest-annotations.md]: All results of AnnotationsHeaders are parsed by macros. It's not that the _filter_ itself is processed, but the entire end result is processed here https://github.com/bndtools/bnd/blob/master/biz.aQute.bndlib/src/aQute/bnd/osgi/AnnotationHeaders.java#L979
In this way, any string contents of any attribute or value can contain macros.

Comment (rotty3000) [reply docs/_chapters/230-manifest-annotations.md]: As @bjhargrave referenced, we have an entire feature built around this behaviour (which _is_ tested) :) Furthermore, additional bnd features beyond bundle annotations use this such as SPI Annotations.

Comment (pkriens) [reply docs/_chapters/230-manifest-annotations.md]: You are right, they are processed properly. But looking at Tim I was not the only one confused.

**Pull request merged**